### PR TITLE
Standardize dev and stage trigger bindings

### PR DESCRIPTION
**Pull request by chetan-rns**
**What type of PR is this?**

> /kind cleanup

**What does this PR do / why we need it**:

- Merge stage-ci-dryrun-from-pr-binding and dev-ci-build-from-pr-binding into ci-build-from-pr-binding
- Merge dev-cd-deploy-from-master-binding and stage-cd-deploy-from-push-binding into cd-deploy-from-push-binding
- Updated trigger templates and event listeners to accommodate new bindings
- Updated test cases

### CODE REVIEW

File: pkg/pipelines/eventlisteners/eventlisteners.go [hunk 1]
```diff
@@ -31,28 +31,28 @@ func Generate(githubRepo, ns, saName string) triggersv1.EventListener {
 					"dev-ci-build-from-pr",
 					devCIBuildFilters,
 					githubRepo,
-					"dev-ci-build-from-pr-binding",
+					"ci-build-from-pr-binding",
```
Comment (bigkevmcd) [inline pkg/pipelines/eventlisteners/eventlisteners.go]: How about we rename this to `github-pr-binding` ?

I think it indicates what it does a bit better (and implies that it's reusable).

File: pkg/pipelines/eventlisteners/eventlisteners.go [hunk 2]
```diff
@@ -31,28 +31,28 @@ func Generate(githubRepo, ns, saName string) triggersv1.EventListener {
...
 					"dev-ci-build-from-pr-template",
 				),
 				createListenerTrigger(
 					"dev-cd-deploy-from-master",
 					devCDDeployFilters,
 					githubRepo,
-					"dev-cd-deploy-from-master-binding",
+					"cd-deploy-from-push-binding",
```
Comment (bigkevmcd) [inline pkg/pipelines/eventlisteners/eventlisteners.go]: And this one might be better as `github-push-binding` ?

File: pkg/pipelines/triggers/bindings.go [hunk 3]
```diff
@@ -52,33 +37,21 @@ func createDevCIBuildBinding(ns string) triggersv1.TriggerBinding {
...
-func createStageCDDeployBinding(ns string) triggersv1.TriggerBinding {
+func createPushBinding(ns string) triggersv1.TriggerBinding {
 	return triggersv1.TriggerBinding{
 		TypeMeta:   triggerBindingTypeMeta,
-		ObjectMeta: meta.ObjectMeta(meta.NamespacedName(ns, "stage-cd-deploy-from-push-binding")),
+		ObjectMeta: meta.ObjectMeta(meta.NamespacedName(ns, "cd-deploy-from-push-binding")),
 		Spec: triggersv1.TriggerBindingSpec{
 			Params: []pipelinev1.Param{
 				createBindingParam("gitref", "$(body.ref)"),
+				createBindingParam("gitreftag", "$(body.head_commit.id)"),
```
Comment (bigkevmcd) [inline pkg/pipelines/triggers/bindings.go]: I think we can possibly do away with this, and change `gitsha` to use `body.head_commit.id` too.

Might wanna check this with a hook notification.

File: pkg/pipelines/eventlisteners/eventlisteners.go (hunk 1 above)
Comment (chetan-rns) [reply pkg/pipelines/eventlisteners/eventlisteners.go]: Renamed to `github-pr-binding`

File: pkg/pipelines/eventlisteners/eventlisteners.go (hunk 2 above)
Comment (chetan-rns) [reply pkg/pipelines/eventlisteners/eventlisteners.go]: Renamed to `github-push-binding`

File: pkg/pipelines/triggers/bindings.go (hunk 3 above)
Comment (chetan-rns) [reply pkg/pipelines/triggers/bindings.go]: Using `gitsha` with value `body.head_commit.id` in github-push-binding

**Pull request merged**